eth/tracers/tracers_test.go

```diff
@@ -82,14 +82,14 @@ func BenchmarkTransactionTrace(b *testing.B) {
 	state := tests.MakePreState(rawdb.NewMemoryDatabase(), alloc, false, rawdb.HashScheme)
 	defer state.Close()
 
-	// Create the tracer, the EVM environment and run it
-	tracer := logger.NewStructLogger(&logger.Config{
-		Debug: false,
-		//DisableStorage: true,
-		//EnableMemory: false,
-		//EnableReturnData: false,
-	})
-	evm := vm.NewEVM(context, state.StateDB, params.AllEthashProtocolChanges, vm.Config{Tracer: tracer.Hooks()})
+	// Create a tracer which records the number of steps
+	var steps = 0
+	tracer := &tracing.Hooks{
+		OnOpcode: func(pc uint64, op byte, gas, cost uint64, scope tracing.OpContext, rData []byte, depth int, err error) {
```

Comment:
I feel like at this point we are benching EVM execution :) it seems this benchmark was introduced in #23016 to test optimizations to the struct logger. What do you think about dropping it?

Reply:
I fixed it instead in the latest commit. It's horribly much slower though, due to the json-encoding and extra mem usage.

Reply:
You were not kidding. It's 10x slower.

PR:

```
goos: darwin
goarch: arm64
pkg: github.com/ethereum/go-ethereum/eth/tracers
BenchmarkTransactionTraceV2-11                15          74619664 ns/op        87275606 B/op     897587 allocs/op
PASS
```

Master:

```
goos: darwin
goarch: arm64
pkg: github.com/ethereum/go-ethereum/eth/tracers
BenchmarkTransactionTrace-11                 163           6710361 ns/op         3798259 B/op      81648 allocs/op
PASS
```

Reply:
I compared the speed on mainnet txes and didn't notice a significant difference. The bench above might be a fluke because we moved the tracer setup now to the benched portion of the test.

Reply:
It is not a "fluke"

> due to the json-encoding 

Previously, the bench collected a bunch of objects. Now we json-encode them. Thus, encoding overhead, plus the fact that memory size of json-encoded data over native size. 

In real life, we would have json-encoded later instead. So it doesn't matter a whole lot.